fs/expose/dokan/__init__.py

```diff
@@ -359,72 +387,72 @@ def _check_lock(self, path, offset, length, info, locks=None):
 
         This method implements basic lock checking.  It checks all the locks
         held against the given file, and if any overlap the given byte range
-        then it returns -ERROR_LOCKED.  If the range is not locked, it will
+        then it returns STATUS_LOCK_NOT_GRANTED.  If the range is not locked, it will
         return zero.
         """
         if locks is None:
             with self._files_lock:
                 try:
                     locks = self._active_locks[path]
                 except KeyError:
-                    return 0
+                    return STATUS_SUCCESS
         for (lh, lstart, lend) in locks:
             if info is not None and info.contents.Context == lh:
                 continue
             if lstart >= offset + length:
                 continue
             if lend <= offset:
                 continue
-            return -ERROR_LOCKED
-        return 0
+            return STATUS_LOCK_NOT_GRANTED
+        return STATUS_SUCCESS
 
     @timeout_protect
     @handle_fs_errors
-    def CreateFile(self, path, access, sharing, disposition, flags, info):
-        path = normpath(path)
+    def ZwCreateFile(self, path, securitycontext, access, attribute, sharing, disposition, options, info):
+        path = self._dokanpath2pyfs(path)
         #  Can't open files that are pending delete.
         if self._is_pending_delete(path):
-            return -ERROR_ACCESS_DENIED
+            return STATUS_ACCESS_DENIED
         # If no access rights are requestsed, only basic metadata is queried.
         if not access:
             if self.fs.isdir(path):
                 info.contents.IsDirectory = True
             elif not self.fs.exists(path):
-                raise ResourceNotFoundError(path)
+                return STATUS_OBJECT_NAME_NOT_FOUND
             return
```

Comment:
You are right it has to return `STATUS_SUCCESS`.
CreateFile for open directory is normal, it is how windows work 😢 

I will try to reproduce the issue you faced. Normally there should be a request with `info.contents.IsDirectory == true` and a create flag.
I don't have my env with me so it can take some times.